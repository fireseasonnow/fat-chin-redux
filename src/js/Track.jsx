import React from 'react';
import styled from 'styled-components';

const Track = ({
    lyricsVisibilityHandler,
    name,
    number,
}) => (
    <Item>
        <Button onClick={() => lyricsVisibilityHandler(name)} type='button'>
            {`${number}. ${name}`}
        </Button>
    </Item>
);

export default Track;

const Item = styled.li`
    position: relative;
    cursor: pointer;
    list-style: none;

    &:hover::after {
        content: 'lyrics';
        font-size: 10px;
        position: absolute;
        right: 0;
        bottom: 0;
    }
`;

const Button = styled.button`
    background: none;
    color: inherit;
    font-size: inherit;
    font-family: inherit;
    border: 0;
    padding: 0;
    cursor: inherit;
    text-align: left;
`;
