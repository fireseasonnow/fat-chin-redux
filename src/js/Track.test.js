import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';

import Track from './Track';

configure({ adapter: new Adapter() });

describe('<Track />', () => {
    let wrapper = {};

    beforeEach(() => {
        wrapper = shallow(<Track />);
    });

    it('should render without crashing', () => {
        expect(wrapper).toBeTruthy();
    });

    it('should render a track with name and number', () => {
        wrapper.setProps({
            number: 1,
            name: 'Hello',
        });

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should fire lyricsVisibilityHandler when clicked on a button', () => {
        const lyricsVisibilityHandler = jest.fn();

        wrapper.setProps({
            name: 'Hello',
            lyricsVisibilityHandler,
        });

        wrapper.find('Track__Button').simulate('click');

        expect(lyricsVisibilityHandler).toHaveBeenCalledWith('Hello');
    });
});
