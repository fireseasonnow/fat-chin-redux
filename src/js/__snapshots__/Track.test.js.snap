// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Track /> should render a track with name and number 1`] = `
<Track__Item>
  <Track__Button
    onClick={[Function]}
    type="button"
  >
    1. Hello
  </Track__Button>
</Track__Item>
`;
